rt/rt.py

```diff
@@ -672,7 +672,10 @@ def edit_ticket(self, ticket_id: typing.Union[str, int], **kwargs: typing.Any) -
         """
         post_data = self.__ticket_post_data(kwargs)
         msg = self.__request('ticket/{}/edit'.format(str(ticket_id)), post_data={'content': post_data})
-        state = msg.split('\n')[2]
+        if msg.count('\n') > 1:
+            state = msg.split('\n')[2]
+        else:
+            state = msg
```

Comment:
I had a similar patch (that I forgot to share, sorry!) that does the following (IIRC the `msg` is empty):

 ````patch
          msg = self.__request('ticket/{}/edit'.format(str(ticket_id)), post_data={'content': post_data})
+        if "" == msg: # Ticket not modified
+            return True
````

Reply:
Would be great to have too

Reply:
Is this better than the proposed solution?

Reply:
Ah, this should be used together, got it! Could you please add it?

Reply:
I think @iamleot solution is better, my request can be removed

Reply:
OK, I was finally able to find a possible reliable use case to reproduce that (unfortunately it's on an internal RT), given `tid` a possible 

````
tracker.edit_ticket(tid, Subject=tracker.get_ticket(tid))
````

...and adding a `breakpoint()` just after the `msg = self.__request('ticket/{}/edit'.format(str(ticket_id)), post_data={'content': post_data})` line:

````
(Pdb) p msg
''
````

In that case I think the patch proposed by @andreyyudin will not work.

@andreyyudin, do you recall how have you catched that? (mostly for completeness/references and/or to spot possible further cases that I am missing)

Reply:
I had that problem (sometimes) when resolving tickets with
```python
self.rt.edit_ticket(ticket_id, Status='resolved')
```

Reply:
Yep the same, it happens when doing edit_ticket(ticket_id, Status='resolved') on an already resolved ticket. The response content in this case is an empty string.

Reply:
Thanks for confirming!

OK if I share the possible patch that I have proposed above as pull request?
(I will also cite @andreyyudin PR's because we have done most of the analysis here)


Thanks again!

Reply:
Sorry for the delay (I had waited before opening PR for possible further feedbacks)!  I have just opened a possible PR that addressed that as discussed.

Reply:
Thanks! Me too, sorry for the delay, merged!